### PR TITLE
add guidance on picking dependencies

### PR DESCRIPTION
**Pull request by jdkandersson**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -51,6 +51,63 @@ This ensures that the tests are run on the same Python version as the charm
...
+  the Canonical internal chat server which holds sensitive data.
+
+The following are helpful indicators in assessing the quality of a dependency:
+
+* How widely adopted the dependency is. For example, `pytest` is widely used
+  across the industry and well maintained reducing the chance of, for example,
+  the project being abandoned or malicous code making it into a release.
+  Indications of adoption can include download counts (https://pypistats.org)
+  and interactions on GitHub such as forks and stars.
+* How well maintained the depenency is. This indicates whether the dependency
```
Comment (merkata) [inline CONTRIBUTING.md:99]: Typo `depenency`.

**Review by merkata — COMMENTED** [review]
Nice addition, I like it, there's one typo - other than that look great!

**Comment by nrobinaubertin** [comment]
LGTM

File: CONTRIBUTING.md (hunk 1 above)
Comment (jdkandersson) [reply CONTRIBUTING.md:99]: Thank you!

**Review by arturo-seijas — APPROVED** [review]
LGTM

**Review by merkata — APPROVED** [review]

**Pull request merged**